ruby/private/download/BUILD.tpl

```diff
@@ -16,6 +16,23 @@ rb_binary(
     main = ":ruby_file",
 )
 
+cc_library(
```

Comment:
nit: there's some flag in bazel forcing these symbols to be `load`ed from a ruleset rather than using the now-deprecated native symbols. If a user turns on that flag I think this will be broken for them.

Reply:
Yeah, we should probably use `rules_cc`: https://blog.bazel.build/2024/12/09/bazel-8-release.html#starlarkification-modularization-of-rules.

Reply:
I've added an explicit dependency on rules_cc and rules_java in https://github.com/bazel-contrib/rules_ruby/pull/197/commits/0f52ccb036b2a5cdeab2f6b9c7551f401748105e, I hope that makes sense. Not sure what is the lowest version I could use so I looked at what other rulesets use.

Reply:
I think this is OK but it feels a bit undesirable that you're forced into carrying these CC/Java toolchain deps even if you have no interest in using them. Don't have any good ideas to avoid it though.

Reply:
That's why I originally didn't add those and used native rules. @alexeagle Can you advise what would be better here?

Reply:
You don't really have a choice. This repo wants to expose the providers for interop with users repository, and providers are symbols rather than strings, so they have to be loaded from the same place.

We'll hope that rules_java and rules_cc are well-maintained so there aren't breaking changes that mean your users are sensitive to the version you pick. In bzlmod you can pick something old, since you're really picking a lower-bound; the MVS algorithm means users can still resolve to something newer.

Reply:
For both rulesets, I used > 1-year-old releases and crossed-check what other dependents do (e.g. rules_python, rules_kotlin).